index.ts

```diff
@@ -1,54 +1,31 @@
 import { multiUrlGenerator } from './lib/multiUrlGenerator';
-import readline from 'readline';
+import fetch from 'node-fetch';
 
-const clientUI = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout,
-});
+// Test fetching Wiki articles
+const requestUrls = multiUrlGenerator(
+  'https://$lang.wikipedia.org/wiki/$article',
+  [
+    { $lang: 'en', $article: 'Plato' },
+    { $lang: 'es', $article: 'Aristóteles' },
+  ]
+);
 
-function createObject(
-  home: string,
-  name: string
-): { $home: string; $name: string } {
-  return {
-    $home: home,
-    $name: name,
-  };
-}
-
-function arrayOfObjectsFactory(wnp: string[]) {
-  //The left side of this array contains the names of the websites and the right side contains the names of the paths.
-  // @ts-ignore
-  const arrayOfObjects: { $home: string; $name: string }[] = [].fill(null);
-  wnp.forEach(function (item: string, index: number) {
-    if (index < wnp.length / 2) {
-      arrayOfObjects.push(
-        createObject(wnp[index], wnp[wnp.length / 2 + index])
+async function main() {
```

Comment:
`async` tells JS that this function will be "`awaiting`" something inside. `await` works as if the asynchronous request (fetching something from another server) was synchronous, so the script "waits" for `fetch` to finish the request, and only after that, it executes the next line.

This is just syntax sugar for a Promise.

https://alligator.io/js/async-functions/

Reply:
👍